### PR TITLE
Add type predicates and implement enlist()

### PR DESCRIPTION
**Pull request by lionelgit**
Following discussion in #12.

I wrote `is_bare_numeric()` to be the bare equivalent of `is.numeric()`, and added `is_bare_double()`. Tell me if you prefer that `is_bare_numeric()` tests for doubles.

### CODE REVIEW

**Comment by hadley** [comment]
I'd prefer that `is_bare_numeric()` tests only for "numeric" vectors, i.e. with `is.double()`.

`is_list()`, `is_bare_list()`, `is_vector()` (`is_list()` or `is_atomic()`) and `is_bare_vector()` would also be useful.

**Comment by hadley** [comment]
FYI the problem with `is.vector()` is that it tests for non-name attributes, not it's behaviour with `NULL`

**Comment by lionelgit** [comment]
ah yes.

And should we have a `is_numeric()` companion for `is_bare_numeric()` then?

What do you have in mind for `is_list()`?

**Comment by hadley** [comment]
Yes to `is_numeric()`, and `is.list()` could just be an alias for `is.list()`

**Comment by hadley** [comment]
I guess we should probably complete the set with `is_character()` , `is_bare_character()`, and  `is_logical()`, `is_bare_logical()`.

File: NAMESPACE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Generated by roxygen2 (4.1.0): do not edit by hand
+# Generated by roxygen2 (4.1.0.9001): do not edit by hand
```
Comment (hadley) [inline NAMESPACE]: Would you mind using the release roxygen2 please?

File: R/objects.R [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+#'
+#' These type predicates aim to make type testing in R more
+#' consistent.
+#'
+#' Most of these predicates are simple aliases to base R functions. In
+#' addition:
+#' \itemize {
+#'   \item Unlike \code{is.atomic()}, \code{is_atomic()} does not
+#'         return \code{TRUE} for \code{NULL}.
+#'   \item Unlike \code{is.vector()}, \code{is_vector()} does not
```
Comment (hadley) [inline R/objects.R]: Maybe `Unlike \code{is.vector()}, \code{is_vector()} test if an object is an atomic vector or a list. \code{is.vector} checks for the presence of attributes (other than name)`

**Comment by hadley** [comment]
Something looks wrong with the generated docs and can you please fix the R CMD check failure?

**Comment by lionelgit** [comment]
Sorry for the roxygen mess, I will fix the other PR.

The doc should be ok now. In the future I'll try to install the repo before submitting.

**Comment by lionelgit** [comment]
roxygen version should be ok in all PR

**Comment by hadley** [comment]
Do you use RStudio?

**Comment by lionelgit** [comment]
I use Emacs

**Comment by hadley** [comment]
Ah, well it's super easy in RStudio, but if you do `devtools::check()` you'll get the same local checks as travis runs

**Comment by lionelgit** [comment]
hmm I do run `check()` with a keybinding... must have missed something

**Comment by hadley** [comment]
I'm still seeing the same failure on travis

**Comment by lionelgit** [comment]
Note to self: run `check()` on a fresh session.

**Comment by lionelgit** [comment]
I'll add: don't confuse `test()` and `check()` ...

**Comment by lionelgit** [comment]
Well this was embarrassing

**Comment by lionelgit** [comment]
Now the problem is that yesterday I did not remove "rbenoulli.Rd" from the git index. As you can tell I'm new to all this.

I removed it from the index in #37 and `check()` does not complain on my machine anymore for this branch so I think it's ok to merge.

I'm in the process of running `check()` on all other branches now.

**Pull request merged**